controllers/deactivation/predicate.go

```diff
@@ -25,3 +29,58 @@ func (CreateAndUpdateOnlyPredicate) Delete(_ event.DeleteEvent) bool {
 func (CreateAndUpdateOnlyPredicate) Generic(_ event.GenericEvent) bool {
 	return false
 }
+
+type GenerationOrConditionsChangedPredicate struct {
+	predicate.GenerationChangedPredicate
+}
+
+// Update implements default UpdateEvent filter for validating no generation change
+func (GenerationOrConditionsChangedPredicate) Update(e event.UpdateEvent) bool {
+	if e.ObjectOld == nil || e.ObjectNew == nil {
+		return false
+	}
+
+	if e.ObjectNew.GetGeneration() != e.ObjectOld.GetGeneration() {
+		return true
+	}
+
+	switch objNew := e.ObjectNew.(type) {
+	case *toolchainv1alpha1.UserSignup:
+		switch objOld := e.ObjectOld.(type) {
+		case *toolchainv1alpha1.UserSignup:
+			if !ConditionsMatch(objOld.Status.Conditions, objNew.Status.Conditions) {
+				return true
+			}
+		}
+	}
+
+	return false
+}
+
+func ConditionsMatch(first, second []toolchainv1alpha1.Condition) bool {
```

Comment:
Looks like a good candidate to move to https://github.com/codeready-toolchain/toolchain-common/blob/master/pkg/condition/condition.go

Reply:
https://github.com/codeready-toolchain/toolchain-common/pull/397